lib/core/firebase/firebase_push_notifications_service.dart

```diff
@@ -1,29 +1,34 @@
+import 'dart:async';
+
 import 'package:firebase_core/firebase_core.dart';
 import 'package:firebase_messaging/firebase_messaging.dart';
 import 'package:hypha_wallet/core/logging/log_helper.dart';
 
-class FirebasePushNotifications {
-  const FirebasePushNotifications._();
-
-  factory FirebasePushNotifications.init() {
-    FirebaseMessaging.instance.getToken().then((value) {
-      // TODO(gguij): Send token to server
-      LogHelper.d('Firebase Device Token: $value');
-    });
+class FirebasePushNotificationsService {
```

Comment:
Removed the factory from this class and its now a regular class with an init method. 
This is because i added a stream to listen when the firebase token changes